### ISSUE TITLE
kpt omits various structured result fields

### ISSUE THREAD

**Issue by prachirp** [GoogleContainerTools/kpt#719]
Running the same function using docker vs using kpt gives different output. Kpt prints part of the results to the console but only prints error message and kubernetes object information. It doesn't print tags fields. This is important because the tags give important information like error code and documentation url.

Steps to reproduce:
`kpt pkg get https://github.com/istio/istio.git/galley/pkg/config/analysis/analyzers/testdata .`

```
cat >fc.yaml <<EOF
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-config
  annotations:
    config.k8s.io/function: |
      container:
        image:  gcr.io/kpt-functions/istioctl-analyze:dev
    config.kubernetes.io/local-config: 'true'
data:
  "flags": [ "--recursive" ]
  "--use-kube": "false"
EOF
```

Compare the output of the below commands:
kpt fn source testdata | docker run -i -u "$(id -u)" --mount type=bind,src="$(pwd)",dst=/source gcr.io/kpt-functions/istioctl-analyze:dev -f /source/fc.yaml | kpt fn sink testdata`
vs.
`kpt fn run testdata --fn-path fc.yaml --results-dir /tmp/`

**Comment by prachirp** [comment]
I find that running the following command puts the same data as the output of `docker run` into `/tmp/results-0.yaml`. This issue is about the UX preference whether kpt should omit tags fields when printing results to the console.
```
kpt fn run testdata --fn-path fc.yaml --results-dir /tmp/
```

**Comment by prachirp** [comment]
kpt omits many fields like `field.path` from structured results, not just tags.

**Comment by prachirp** [comment]
We have decided to add field.path but omit tags when printing to console

**Comment by prachirp** [comment]
released kpt-functions 0.14.5 which adds field.path to console output
